### PR TITLE
Add unit option to Indirection Diffraction UI output plot options

### LINKED ISSUE #33551: Offer plotting in QSpace in Indirect Diffraction
**Is your feature request related to a problem? Please describe.**
When using Indirect Diffraction GUI there is the option to plot a spectrum after successfully loading a run on diffspec. At the moment the plot produced is in d-space. Users would like the option to have the results in Q-Space.

**Describe the solution you'd like**
Provide users with the ability to choose whether their plot is in d-space or Q-Space and produce the relevant plot.

**Additional context**
Requested by Users

### PR DESCRIPTION
**Pull request by MohamedAlmaki**
**Description of work.**
This PR adds an option to select the unit of the output plot in the indirect diffraction interface. Users can now choose from "D-spacing" or "Q-spacing" unit options. 

**To test:**
1- Open Indirect Diffraction interface (Interfaces->Indirect->Diffraction).
2- Change the Reflection setting to diffspec.
3- Click the Run button. 
4- Once the run is complete, verify that the unit combo box and the `Plot Spectra` button are both enabled.
5- Select the two possible units in the unit combo box and plot the spectra for each unit.
6- Verify that the plot displays the correct unit in every case.
7- Enable the `Use Manual Grouping` option and repeat the above steps.
8- Verify that the behavior of the Plot Spectra option remains the same with manual grouping enabled.

Fixes #33551. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by MohamedAlmaki** [comment]
[Add unit option to Indirection Diffraction UI output plot options](https://trello.com/c/6SRwH7vp/9-add-unit-option-to-indirection-diffraction-ui-output-plot-options)

**Comment by MialLewis** [comment]
When trying to `Run` with `use Manual Grouping` I get this error:

 `Error in execution of algorithm GroupDetectors:
All arguments to init must be positive and non-zero
Error in execution of algorithm ISISIndirectDiffractionReduction
All arguments to init must be positive and non-zero
  at line 320 in 'C:/Repos/GitHub/conda/mantid_pr_review/Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ISISIndirectDiffractionReduction.py'
  caused by line 666 in 'C:\Repos\GitHub\conda\mantid_pr_review\scripts\Inelastic\IndirectReductionCommon.py'
  caused by line 756 in 'C:\Repos\GitHub\conda\mantid_pr_review\scripts\Inelastic\IndirectReductionCommon.py'`
  
This is fixed by setting number of groups to `1`. Can we handle this error more gracefully? If the number of groups cannot be zero should this be the default?

Appreciate this one may not be related to your code changes in this PR.

**Review by MialLewis — CHANGES_REQUESTED** [review]
A couple of issues with functionality to look at.

**Comment by MohamedAlmaki** [comment]
> When trying to `Run` with `use Manual Grouping` I get this error:
> 
> `Error in execution of algorithm GroupDetectors: All arguments to init must be positive and non-zero Error in execution of algorithm ISISIndirectDiffractionReduction All arguments to init must be positive and non-zero at line 320 in 'C:/Repos/GitHub/conda/mantid_pr_review/Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ISISIndirectDiffractionReduction.py' caused by line 666 in 'C:\Repos\GitHub\conda\mantid_pr_review\scripts\Inelastic\IndirectReductionCommon.py' caused by line 756 in 'C:\Repos\GitHub\conda\mantid_pr_review\scripts\Inelastic\IndirectReductionCommon.py'`
> 
> This is fixed by setting number of groups to `1`. Can we handle this error more gracefully? If the number of groups cannot be zero should this be the default?
> 
> Appreciate this one may not be related to your code changes in this PR.

Yes, this happens because I modify the workspace in place. I'll fix this

**Comment by MialLewis** [comment]
The first bug is now fixed - thanks. 

If I click `run` with `use Manual Grouping` I currently see 
image


Can you replicate this?

File: qt/scientific_interfaces/Indirect/IndirectPlotOptionsView.cpp [hunk 1]
```diff
@@ -68,6 +68,9 @@ void IndirectPlotOptionsView::setupView() {
   connect(m_plotOptions->cbWorkspace, SIGNAL(currentIndexChanged(QString const &)), this,
           SLOT(emitSelectedWorkspaceChanged(QString const &)));
 
+  connect(m_plotOptions->cbPlotUnit, SIGNAL(currentIndexChanged(QString const &)), this,
```
Comment (MialLewis) [inline qt/scientific_interfaces/Indirect/IndirectPlotOptionsView.cpp:71]: It looks like this function has been made obsolete: https://doc.qt.io/qt-5/qcombobox-obsolete.html

Can we change to use this in this file? https://doc.qt.io/qt-5.15/qcombobox.html#currentIndexChanged

**Review by MialLewis — CHANGES_REQUESTED** [review]
Looks good, just one comment regarding an obsolete `qt` function which you may want to switch out if it's not too much effort.

In addition, I suggest raising an issue regarding the error I encountered having an incorrect instrument - the error should handled and stated explicitly given it's so easy to encounter.

File: qt/scientific_interfaces/Indirect/IndirectPlotOptionsView.cpp (hunk 1 above)
Comment (MohamedAlmaki) [reply qt/scientific_interfaces/Indirect/IndirectPlotOptionsView.cpp]: > Looks good, just one comment regarding an obsolete `qt` function which you may want to switch out if it's not too much effort.
> 
> In addition, I suggest raising an issue regarding the error I encountered having an incorrect instrument - the error should handled and stated explicitly given it's so easy to encounter.

The error happens in an algorithm so I think printing the error in the log makes sense. What is the better way to state the error? Is it by showing a pop-up menu?

Comment (MialLewis) [reply qt/scientific_interfaces/Indirect/IndirectPlotOptionsView.cpp]: If it isn't easy to catch the exception and then print a clearer message to the log, perhaps we could add validation with regards to the instrument before the algorithm is executed.

The error message currently output to log is cryptic, it isn't immediately clear that a user could do something so simple as changing the instrument to correct it.

If this error existed before this PR I'd raise another issue for this.

Comment (MohamedAlmaki) [reply qt/scientific_interfaces/Indirect/IndirectPlotOptionsView.cpp]: ok, thanks. I've raised a new issue here #35480

**Comment by MialLewis** [comment]
When I plot a contour with the units specified as `Q-spacing`, the x units are still `d-spacing`. Is this intended?

image

**Comment by MohamedAlmaki** [comment]
> When I plot a contour with the units specified as `Q-spacing`, the x units are still `d-spacing`. Is this intended?
> 
> image

Nope, I will check this

**Review by MialLewis — APPROVED** [review]
This works well now, thanks for the changes.

**Comment by sf1919** [comment]
@mantidproject/gatekeepers Adding screenshot to show all tests approved and PR approved
Screenshot (264)

**Pull request merged**